### ISSUE TITLE
Repository list is incomplete

### ISSUE THREAD

**Issue by vyazelenko** [ben-manes/gradle-versions-plugin#90]
I have this simple `build.gradle` file:
```groovy
buildscript {
    repositories {
        maven { 
		    url "https://maven.java.net/content/groups/public" 
        }
    }
}

plugins {
    id 'java'
    id 'com.github.ben-manes.versions' version '0.11.3'
}

repositories {
    maven { 
	    url "https://repo1.maven.org/maven2" 
    }
}

dependencies {
    compile "ai.h2o:h2o-core:3.6.0.11"
}
```
When I execute `gradle dU --info` I get the following output:
```bash
Starting Build
Settings evaluated using settings file 'P:\Projects\OpenSource\gradle-versions-multi-repos\settings.gradle'.
Projects loaded. Root project using build file 'P:\Projects\OpenSource\gradle-versions-multi-repos\build.gradle'.
Included projects: [root project 'gradle-versions-multi-repos']
Evaluating root project 'gradle-versions-multi-repos' using build file 'P:\Projects\OpenSource\gradle-versions-multi-repos\build.gradle'.
All projects evaluated.
Selected primary task 'dependencyUpdates' from project :
Tasks to be executed: [task ':dependencyUpdates']
:dependencyUpdates (Thread[main,5,main]) started.
:dependencyUpdates
Executing task ':dependencyUpdates' (up-to-date check took 0.0 secs) due to:
  Task has not declared any outputs.
Resolving with repositories:
 - maven: https://maven.java.net/content/groups/public
 - maven2: https://plugins.gradle.org/m2
Comparing dependency (current: ai.h2o:h2o-core:3.6.0.11, latest: 3.6.0.11)
Comparing dependency (current: com.github.ben-manes:gradle-versions-plugin:0.11.3, latest: 0.11.3)

------------------------------------------------------------
: Project Dependency Updates (report to plain text file)
------------------------------------------------------------

The following dependencies are using the latest milestone version:
 - ai.h2o:h2o-core:3.6.0.11
 - com.github.ben-manes:gradle-versions-plugin:0.11.3

All dependencies are using the latest milestone versions.

Generated report file build/dependencyUpdates/report.txt
:dependencyUpdates (Thread[main,5,main]) completed. Took 0.327 secs.
```
Note that repositories list is **incomplete** as repository from `repositories` section is missing. The reason for it is that this repository has name "maven" which is also the name of the repository from `buildscript` block.

The workaround is to add name to one of the repositories like:
```groovy
buildscript {
    repositories {
        maven {
            name "repo 1"		
		    url "https://maven.java.net/content/groups/public" 
        }
    }
}

plugins {
    id 'java'
    id 'com.github.ben-manes.versions' version '0.11.3'
}

repositories {
    maven { 
        name "repo 2"		
	    url "https://repo1.maven.org/maven2" 
    }
}

dependencies {
    compile "ai.h2o:h2o-core:3.6.0.11"
}
```
And then repositories list is complete:
```bash
...
Executing task ':dependencyUpdates' (up-to-date check took 0.0 secs) due to:
  Task has not declared any outputs.
Resolving with repositories:
 - repo 1: https://maven.java.net/content/groups/public
 - maven: https://plugins.gradle.org/m2
 - repo 2: https://repo1.maven.org/maven2
...
```

**Comment by ben-manes** [comment]
Sorry, this was also found in #66. I haven't been actively working on the plugin so your workaround is usually my first suggestion to users.

I think this problem only began happening in the late Gradle 2.x series, but I have not backtracked to verify. We are aggregating repositories for convenience by using Gradle's `RepositoryHandler` to add them to a configuration. Unfortunately that is very buggy, because Gradle tries to remove duplicates but mistakenly ignores additions. We had used `addAll` in an earlier versions, but that made the build crawl because it allows duplicates and resolves them repeatedly in large builds. We were not able to get the Gradle team to acknowledge the [issue](https://discuss.gradle.org/t/repository-duduping/10327/3) when we figured out the problem.

The only reliable fix is to remove aggregation and use whatever repositories are defined for a configuration. This has the added benefit of not touching `RepositoryHandler` and not querying Gradle's plugin repository for compile artifacts.
